snuba_sdk/mql_visitor.py

```diff
@@ -105,14 +102,17 @@ def _combine(
         }
 
     def _visit_query(
-        self, query: Timeseries | Formula | None
+        self, query: Timeseries | Formula | str | None
     ) -> Mapping[str, str | Mapping[str, str]]:
         if query is None:
             raise InvalidMetricsQueryError("MetricQuery.query must not be None")
```

Comment:
why is None a valid type in the type signature if it'll just throw an exception?

Reply:
Because we want to allow people to instantiate these objects without exceptions being thrown. This allows people to build these objects iteratively. This check happens when we actually serialize the object, since at that point we need a value here.